clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -420,7 +420,7 @@ private static int parseAcks(String acksString) {
      *
      * @throws InterruptException If the thread is interrupted while blocked
      * @throws SerializationException If the key or value are not valid objects given the configured serializers
-     * @throws BufferExhaustedException If <code>block.on.buffer.full=false</code> and the buffer is full.
+     * @throws TimeoutException if the time taken for fetching metadata, serialization of (key,value), partitioning has surpassed <code>max.block.ms</code>.
```

Comment:
@MayureshGharat, it was removed because the performance impact was unacceptable, the JIRA has more details:
https://issues.apache.org/jira/browse/KAFKA-2950